src/windows/disk.rs

```diff
@@ -383,6 +374,63 @@ fn os_string_from_zero_terminated(name: &[u16]) -> OsString {
     OsString::from_wide(&name[..len])
 }
 
+fn get_disk_kind(device_path: &[u16], borrowed_handle: &Option<HandleWrapper>) -> DiskKind {
+    let binding = (
+        borrowed_handle,
+        if borrowed_handle.is_none() {
+            unsafe { HandleWrapper::new_from_file(device_path, Default::default()) }
+        } else {
+            None
+        },
+    );
+    let handle = match binding {
+        (Some(ref handle), _) => handle,
+        (_, Some(ref handle)) => handle,
+        (None, None) => return DiskKind::Unknown(-1),
+    };
+
+    if handle.is_invalid() {
+        sysinfo_debug!(
+            "Expected handle to '{:?}' to be valid",
+            String::from_utf16_lossy(device_path)
+        );
+        return DiskKind::Unknown(-1);
+    }
+
+    let spq_trim = STORAGE_PROPERTY_QUERY {
+        PropertyId: StorageDeviceSeekPenaltyProperty,
+        QueryType: PropertyStandardQuery,
+        AdditionalParameters: [0],
+    };
+    let mut result: DEVICE_SEEK_PENALTY_DESCRIPTOR = unsafe { std::mem::zeroed() };
+
+    let mut dw_size = 0;
+    let device_io_control = unsafe {
+        DeviceIoControl(
+            handle.0,
+            IOCTL_STORAGE_QUERY_PROPERTY,
+            Some(&spq_trim as *const STORAGE_PROPERTY_QUERY as *const c_void),
+            size_of::<STORAGE_PROPERTY_QUERY>() as u32,
+            Some(&mut result as *mut DEVICE_SEEK_PENALTY_DESCRIPTOR as *mut c_void),
+            size_of::<DEVICE_SEEK_PENALTY_DESCRIPTOR>() as u32,
+            Some(&mut dw_size),
+            None,
+        )
+        .is_ok()
+    };
+
+    if !device_io_control || dw_size != size_of::<DEVICE_SEEK_PENALTY_DESCRIPTOR>() as u32 {
```

Comment:
```suggestion
        if !device_io_control || dw_size as usize != size_of::<DEVICE_SEEK_PENALTY_DESCRIPTOR>() {
```

Less possible to have a cast error in case something changes in `windows-rs`.